pkg/apis/v1alpha1/networkinfo_types.go

```diff
@@ -0,0 +1,48 @@
+/* Copyright © 2024 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+//+kubebuilder:object:root=true
+//+kubebuilder:storageversion
+
+// NetworkInfo is used to report the network info for a namespace.
+type NetworkInfo struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	VPCs []VPCState `json:"vpcs"`
+}
+
+//+kubebuilder:object:root=true
+
+// NetworkInfoList contains a list of NetworkInfo.
+type NetworkInfoList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []NetworkInfo `json:"items"`
+}
+
+// VPCState defines information for VPC.
+type VPCState struct {
+	// VPC name.
+	Name       string      `json:"name"`
+	Conditions []Condition `json:"conditions"`
+	// NSX Policy path for VPC.
+	VPCPath string `json:"vpcPath"`
+	// Default SNAT IP for Private Subnets.
+	DefaultSNATIP string `json:"defaultSNATIP"`
+	// LB IPs(AVI SE Subnet CIDR or NSX LB SNAT IP).
+	LBIPs []string `json:"lbIPs,omitempty"`
```

Comment:
Better to use full name "LoadBalancerIPAddresses". Do we need a slice?

Reply:
ok, changed to LoadBalancerIPAddresses. For NSX LB VIPs, I am thinking we might have multiple LB services. Make sense?

Reply:
I am not very against to support multiple IPs, but I do not think we need to include VIPs here. As we discussed offline, for that special case, let us assume users can know the VIPs.

Reply:
But do you think we can support multiple addresses with a single string - IPs are separated by ","?

Reply:
Yea, I think we can support multiple addresses with a single string - IPs are separated by ",".